### ISSUE TITLE
Bug: Excessive console messages with ConstraintWeightOverride

### ISSUE THREAD

**Issue by grizzleeadam** [TimefoldAI/timefold-solver#1131]
**Describe the bug**
When using ConstraintWeightOverrides with Timefold 1.14.0, the constraint weights seem to be updated every couple of moves during solving, even when no constraint weights have been changed. My program only reads the configured weights once when a file is loaded, I'm not sure why constraint weights are being re-set so frequently.

**Expected behavior**
Constraint weights should only be updated when setConstraintWeightOverrides is called, not routinely during solving.

**Actual behavior**
Constraint weights are updated when setConstraintWeightOverrides is called, then again after every couple solver steps. My program executes ~100-200 steps per second, and the console log is absolutely filled with constraint weights.

**To Reproduce**
Link to a small reproducer or attach an archive containing the reproducer to the issue.
Alternatively, provide clear and concise steps to reproduce the behavior.

## Environment

**Timefold Solver Version or Git ref**:  1.140.0

**Output of `java -version`:** 
openjdk version "21.0.3" 2024-04-16 LTS
OpenJDK Runtime Environment Temurin-21.0.3+9 (build 21.0.3+9-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (build 21.0.3+9-LTS, mixed mode, sharing)

**Output of `uname -a` or `ver`:**

## Additional information

Provide any and all other information which might be relevant to the issue.

**Comment by Christopher-Chianelli** [comment]
Duplicate of https://github.com/TimefoldAI/timefold-solver/issues/1076 ; Resolved in `main` via https://github.com/TimefoldAI/timefold-solver/pull/1091. Will be fixed in the 1.15.0 release.

**Comment by grizzleeadam** [comment]
It seems this is called every time a "new best score" is found, at least in my application. Is there a way to hide this particular message without disabling Debug logging?

**Comment by triceo** [comment]
@grizzleeadam The code is triggered every time a new scoring session is created. This typically happens when a new solver starts, and _not_ when new best solution is found. I do not understand how you can be getting the behavior you are describing - can you please share a reproducer?

**Comment by grizzleeadam** [comment]
@triceo My application started as an OptaPlanner example, which was eventually migrated to Timefold (but still relies on much of the deprecated SwingUI framework from OptaPlanner, such as SolutionBusiness and SolutionXlsxFileIO, etc). I wonder if some of that is convoluting the issue.

**Comment by triceo** [comment]
Probably. It's been a while since I last saw the old example app, but if my memory serves me, there was code that would trigger some solver operations on every new best solution. Let's say that the app wasn't the pinnacle of software design. :-)

**Comment by grizzleeadam** [comment]
That makes sense. I’ve noticed that the issue does not appear when running the Benchmarker, likely because it bypasses many of those extraneous operations. (Similarly, the Benchmarker fails when loading an initialized solution when shadow variables are involved…)

I’m working on rebuilding into a Quarkus application, but all the new frameworks have been a steep learning curve - it’s coming along slowly but surely. 

Thanks for the insights.

**Comment by grizzleeadam** [comment]
> Probably. It's been a while since I last saw the old example app, but if my memory serves me, there was code that would trigger some solver operations on every new best solution. Let's say that the app wasn't the pinnacle of software design. :-)

I'm glad you mentioned this. Digging through the code, I found that the GUI was calling solutionManager.update(...) for refreshing the displayed Score every time a new solution was found. 

By refactoring this code, the duplicate calls were fixed, the program is faster AND I'm no longer getting the random shadow variable corruption issues I was having.

Thanks for pointing me in the right direction!
